Gems/Atom/RPI/Code/Source/RPI.Public/Pass/ParentPass.cpp

```diff
@@ -479,6 +513,74 @@ namespace AZ
             return nullptr;
         }
 
+        void ParentPass::ClearMergeAsSubpassesFlag()
+        {
+            m_flags.m_mergeChildrenAsSubpasses = false;
+            for (auto& childPass : m_children)
+            {
+                childPass->m_flags.m_mergeChildrenAsSubpasses = false;
+            }
+        }
+
+        void ParentPass::CreateRenderAttachmentConfigurationForSubpasses()
+        {
+            if (!m_flags.m_mergeChildrenAsSubpasses)
+            {
+                // This is the most common scenario.
+                return;
+            }
+
+            // We validated already during AddChild() and InsertChild() that all children are
+            // of type RasterPass or subclass of RasterPass.
+            const auto numSubpasses = m_children.size();
+            AZ_Assert(numSubpasses > 1, "A parent pass that merges its children must have at least two children");
```

Comment:
Might be kinder to the user if this were an Error + early out instead of an Assert?

Reply:
Fixed. Changed to AZ_Error